### PR TITLE
feat(prompts): Add prompt combobox to playground page + deeplink to prompt specific playground

### LINKED ISSUE #5735: [hub] [ui] prompt id and version in zustand state

### LINKED ISSUE #5736: [hub] [ui] prompt picker / combobox in playground

### PR DESCRIPTION
**Pull request by cephalization**
https://github.com/user-attachments/assets/adda9456-482a-4d03-a238-c4a85ca4c22e

Resolves #5736 
Resolves #5735

### CODE REVIEW

File: app/src/pages/playground/Playground.tsx [hunk 1]
```diff
@@ -214,7 +214,7 @@ const playgroundInputOutputPanelContentCSS = css`
  * This width accomodates the model config button min-width, as well as chat message accordion
  * header contents such as the chat message mode radio group for AI messages
  */
-const PLAYGROUND_PROMPT_PANEL_MIN_WIDTH = 475;
+const PLAYGROUND_PROMPT_PANEL_MIN_WIDTH = 570;
```
Comment (cephalization) [inline app/src/pages/playground/Playground.tsx:217]: Without this bump, the chat template delete button will get cut off on small screen sizes now that the prompt selector takes up some of the header's room

File: app/src/pages/playground/PlaygroundChatTemplate.tsx [hunk 2]
```diff
@@ -377,30 +377,39 @@ function SortableMessageItem({
...
+                  instanceId: playgroundInstanceId,
+                  patch: {
+                    template: {
+                      __type: "chat",
+                      messages: template.messages.map((msg) =>
+                        msg.id === message.id
+                          ? { ...msg, role, toolCalls }
+                          : msg
+                      ),
+                    },
```
Comment (cephalization) [inline app/src/pages/playground/PlaygroundChatTemplate.tsx:407]: Only indentation changed here

File: app/src/pages/prompt/promptLoader.tsx [hunk 3]
```diff
@@ -26,5 +26,7 @@ export async function promptLoader(args: LoaderFunctionArgs) {
     {
       id: promptId as string,
     }
-  ).toPromise();
+  )
+    .toPromise()
+    .catch(() => new Response("Prompt not found", { status: 404 }));
```
Comment (cephalization) [inline app/src/pages/prompt/promptLoader.tsx:31]: This displays a human readable error message when you enter a bad slug. We should upgrade the ErrorComponent to display a Not Found page if the error is instance of Response and has status 404

File: app/src/pages/playground/PromptComboBox.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    useQueryLoader<PromptComboBoxQuery>(promptComboBoxQuery);
+
+  useEffect(() => {
+    // TODO(apowell): Paginate and filter in query
+    loadQuery({ first: 100 });
+    return () => disposeQuery();
+  }, [disposeQuery, loadQuery]);
+
+  return queryReference != null ? (
+    <PromptComboBoxLoader queryReference={queryReference} {...props} />
```
Comment (mikeldking) [inline app/src/pages/playground/PromptComboBox.tsx:87]: I think the naming is a bit reversed for me where this component is the loader and the content is the combox itself. I would probably maybe create a loader and the content and then expose the composition of the two as the  combobox component.

File: app/src/pages/playground/PromptComboBox.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+type PromptComboBoxProps = {
+  onChange: (promptId: string | null) => void;
+  promptId?: string | null;
+  container?: HTMLElement;
+} & Omit<
+  ComboBoxProps<PromptItem>,
+  "children" | "onSelectionChange" | "defaultSelectedKey"
+>;
+
+export function PromptComboBoxLoader({
```
Comment (mikeldking) [inline app/src/pages/playground/PromptComboBox.tsx:27]: no real reason to export

File: app/src/pages/playground/PromptComboBox.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+          __typename
+          ... on Prompt {
+            id
+            name
+          }
+        }
+      }
+    }
+  }
+`;
```
Comment (mikeldking) [inline app/src/pages/playground/PromptComboBox.tsx:105]: suggetion: I find it a bit more intuitive to just have the gql DSL right inline

**Review by mikeldking — APPROVED** [review]

File: app/src/pages/playground/PromptComboBox.tsx (hunk 4 above)
Comment (cephalization) [reply app/src/pages/playground/PromptComboBox.tsx:87]: Good call. Was just copying convention from the other combobox file.

File: app/src/pages/playground/PromptComboBox.tsx (hunk 6 above)
Comment (cephalization) [reply app/src/pages/playground/PromptComboBox.tsx:105]: Agree. Will change

**Pull request merged**